libmamba/src/api/install.cpp

```diff
@@ -933,7 +933,13 @@ namespace mamba
                     std::vector<std::string> f_specs;
                     for (auto& line : file_contents)
                     {
-                        if (line[0] != '#' && line[0] != '@')
+                        // Skip comment lines and empty lines
+                        // Comment lines start with '#' or '@' preceded by whitespaces or tabs
+                        auto is_comment = util::starts_with(line, "#")
+                                          || util::starts_with(line, "@");
+                        auto is_empty = std::all_of(line.begin(), line.end(), ::isspace);
```

Comment:
I would prefer that we use `std::isspace` or `mamba::isspace` but not global `isspace`. It's the same as `std::isspace` but we avoid potential surprises by being explicit.